platform/eventing/listener.go

```diff
@@ -0,0 +1,74 @@
+package eventing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	cloudeventsnats "github.com/cloudevents/sdk-go/v2/protocol/nats"
+	"github.com/jmoiron/sqlx"
+)
+
+type listenerSubject string
+
+// Listener defines what a nats listener looks like
+type Listener struct {
+	URL      string
+	subjects []string
+	db       *sqlx.DB
+	logger   *log.Logger
+}
+
+// NewListener creates a new config for a listener
+func NewListener(URL string, subjects []string, db *sqlx.DB, logger *log.Logger) *Listener {
+	return &Listener{
+		URL:      URL,
+		subjects: subjects,
+		db:       db,
+		logger:   logger,
+	}
+}
+
+// Start starts a listener for the provided subjects
+// It will save all events for the subjects in the DB
+func (listener *Listener) Start() {
+	for _, subject := range listener.subjects {
+		ctx := context.WithValue(context.Background(), listenerSubject(subject), subject)
+
+		natsConsumer, err := cloudeventsnats.NewConsumer(listener.URL, subject, cloudeventsnats.NatsOptions())
+		if err != nil {
+			listener.logger.Printf("failed to create nats consumer, %v", err)
+		}
```

Comment:
should we crash on this one ? It could mean that there is a configuration mismatch somewhere